src/web/routes/entities-v2/entities-router.ts

```diff
@@ -0,0 +1,144 @@
+import { HttpStatusCode } from 'axios';
+import type { NextFunction } from 'express';
+import { Router } from 'express';
+
+import {
+	GET_ENTITY_BY_URL_REQUEST_SCHEMA,
+	ON_ENTITY_ASSOCIATED_REQUEST_SCHEMA,
+	ON_ENTITY_DISASSOCIATED_REQUEST_SCHEMA,
+} from './schemas';
+import type {
+	GetEntityByUrlRequest,
+	GetEntityByUrlResponse,
+	OnEntityAssociatedRequest,
+	OnEntityAssociatedResponse,
+	OnEntityDisassociatedRequest,
+	OnEntityDisassociatedResponse,
+} from './types';
+
+import { tryParseUrl } from '../../../common/url-utils';
+import {
+	getDesignByUrlForBackfillUseCase,
+	getDesignByUrlUseCase,
+	onDesignAssociatedWithIssueUseCaseParams,
+	onDesignDisassociatedFromIssueUseCase,
+} from '../../../usecases';
+import { BadRequestResponseStatusError } from '../../errors';
+import { requestSchemaValidationMiddleware } from '../../middleware';
+import { jiraServerToServerSymmetricJwtAuthenticationMiddleware } from '../../middleware/jira';
+
+export const entitiesRouterV2 = Router();
+
+entitiesRouterV2.use(jiraServerToServerSymmetricJwtAuthenticationMiddleware);
+
+/**
+ * Returns Figma design details for the given URL.
+ * Invoked when an entity is being associated with an Atlassian resource (e.g. a Jira Issue).
+ *
+ * @see https://developer.atlassian.com/cloud/devops-provider-actions/specifications/current/#get-entity-by-url
+ */
+entitiesRouterV2.put(
+	'/getEntityByUrl',
+	requestSchemaValidationMiddleware(GET_ENTITY_BY_URL_REQUEST_SCHEMA),
+	(
+		req: GetEntityByUrlRequest,
+		res: GetEntityByUrlResponse,
+		next: NextFunction,
+	) => {
+		const { connectInstallation } = res.locals;
+		const atlassianUserId = req.query.userId;
+
+		const designUrl = tryParseUrl(req.body.entity.url);
+
+		if (!designUrl) {
+			return next(new BadRequestResponseStatusError('Invalid design URL.'));
+		}
+
+		// TODO: The backfill implementation is temporary. Delete it once it is deprecated.
+		// Determine whether the design is being backfilled. The design is considered for backfill if it contains a
+		// special query parameter (`com.atlassian.designs.backfill`).
+		// Jira Frontend appends this query parameter to the design URL as a temporary workaround that allows to
+		// distinguish "Backfill" from normal "Associate" operations. Therefore, the app can apply some special considerations
+		// to designs for backfill (e.g., handle deleted designs).
+		const isBackfill = designUrl.searchParams.has(
+			'com.atlassian.designs.backfill',
+			'true',
+		);
+
+		if (isBackfill) {
+			getDesignByUrlForBackfillUseCase
```

Comment:
To clarify: the Figma app will return the minimal design for the backfill use case which will be ingested with the issue association by Jira and the app will then async send the full design (without the issue association). This is temporary until the backfill use case can be removed after the Figma widget is updated.